### DISCUSSION TITLE
On / Off Support Not Working

### DISCUSSION THREAD

**Discussion by faisalhash** [osk2/yamaha-soundbar#3] [Q&A]
Great contribution! I managed to get it working on my YAS 209.

Is there a way where I can add turn on / turn off support? I keep getting media_player.turn_on / turn_off not supported by this device whenever I try to manually toggle the power button on my dashboard.

**Comment by osk2** [comment]
As you may know, this component was forked from [nagyrobi/home-assistant-custom-components-linkplay](https://github.com/nagyrobi/home-assistant-custom-components-linkplay) which doesn't implement this feature.

The app relies on an API called "power saving" to switch power on and off. However, I've decided not to include this feature because it appears that the API is not functioning as we expected.

**Comment by LindyMan93** [comment]
My work around for turning the soundbar on was just to play media through it then switch the source.

**Comment by siremarcus** [comment]
Hello, I really wish there was an ON/OFF action available.
I also found a way of turning my ATS-2090 on by switching between bluetooth and HDMI sources, with a 10 second timer in between.
Did not found a way to turn off the device though.

In any case, thank you for this addon

**Comment by osk2** [comment]
I think I've found the way to implement on/off switch. I’ll run some tests and hopefully get it done by the end of the month. 💪

**Reply by idodror** [reply]
hey,
really looking forward to it!
there is an update? :)

tnx

**Reply by osk2** [reply]
Sorry, no update yet. I've been swamped. Will find some time to take a look.

**Reply by idodror** [reply]
> Sorry, no update yet. I've been swamped. Will find some time to take a look.

Thank you for the comment.
Instead of it, can you explain how to get the power saving status? I can try to put it in the code by myself.
I managed to set this parameter with "yamaha_soundbar.sound_settings" but not to check it's current status..

Can you help?

**Reply by osk2** [reply]
I've added turn_on/turn_off in the `feature/power_switch` branch, but I can't get the state to update after switching the power. Anyway, contributions are welcome if you're interested.
